kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -79,7 +79,18 @@ class BatchJobSubmission(
   def appStartTime: Long = _appStartTime
   def appStarted: Boolean = _appStartTime > 0
 
-  private lazy val _submitTime = if (appStarted) _appStartTime else System.currentTimeMillis
+  private lazy val _submitTime = System.currentTimeMillis
+  private lazy val waitEngineCompletion = builder.waitEngineCompletion
+  private def submitTime: Long = if (appStarted) {
```

Comment:
could you add comments to clarify the meaning of `appStartTime`, `appStarted`, `submitTime`?